dotnet_grpc_bench/GreeterServer/Program.cs

```diff
@@ -16,37 +16,45 @@
 
 #endregion
 
-using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using GreeterServer.Services;
+using GreeterServer;
+using System;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.WebHost.ConfigureKestrel(options =>
+{
+	options.AddServerHeader = false;
+	options.ListenAnyIP(50051, listenOptions =>
+	{
+		listenOptions.Protocols = HttpProtocols.Http2;
+	});
+});
 
-namespace GreeterServer
+builder.WebHost.ConfigureLogging(logging =>
 {
-    public class Program
-    {
-        public static void Main(string[] args)
-        {
-            CreateHostBuilder(args).Build().Run();
-        }
-
-        public static IHostBuilder CreateHostBuilder(string[] args) =>
-            Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.ConfigureKestrel(options =>
-                    {
-                        options.AddServerHeader = false;
-                        options.ListenAnyIP(50051, listenOptions =>
-                        {
-                            listenOptions.Protocols = HttpProtocols.Http2;
-                        });
-                    });
-                    webBuilder.ConfigureLogging(logging =>
-                    {
-                        logging.ClearProviders();
-                    });
-                    webBuilder.UseStartup<Startup>();
-                });
-    }
-}
+	logging.ClearProviders();
+});
+
+builder.Services.AddGrpc(o => o.IgnoreUnknownServices = true);
+builder.Services.Configure<RouteOptions>(c => c.SuppressCheckForUnhandledSecurityMetadata = true);
+builder.Services.AddSingleton<GreeterService>();
+
+await using var app = builder.Build();
```

Comment:
Actually, one weird thing. What is the deal with this await? Is Build is async?

@halter73 @davidfowl

Reply:
Oh, never mind. It's for DisposeAsync.

Reply:
Right. Though it's arguably unnecessary since RunAsync disposes internally. We're thinking of removing the `await using` from the template.

Reply:
If it's not required I'll update the PR by removing the using await.

Reply:
@JamesNK Ok I've updated the PR and removed the using await. This might be a better discussion to have at dotnet/aspnetcore but woudnt it be better if WebApplicationBuilder Build method return an interface of IWebApplication whithout the IAsyncDisposable interface exposed to the comsumer if its normally not expected to be disposed?

Reply:
RunAsync handles dispose so it isn't needed here. You can also use StartAsync to start the web app, in which case you want StopAsync/DisposeAsync.

Reply:
Ah ok. That makes sense then.